[KOGITO-7481] - [KWS-Guides] Document the Enhanced Service Discovery (#177)
kiegroup/kogito-docs: serverlessworkflow/modules/ROOT/pages/index.adoc

Learn how to build images for your workflow applications using Quarkus CLI
--

[.card]
--
[.card-title]

> xref:cloud/enhanced-service-discovery.adoc[Enhanced Service Discovery]

Reviewer: ```suggestion
xref:cloud/enhanced-service-discovery.adoc[Kubernetes Service Discovery for Workflow Application Configuration]
```
Also change the file name please.

> xref:cloud/kubernetes-service-discovery.adoc[Kubernetes Service Discovery for Serverless Workflow Application Configuration]

Reviewer 2: ```suggestion
xref:cloud/kubernetes-service-discovery.adoc[Kubernetes Service Discovery for Serverless Workflow Application]
```

[.card-description]

> Learn what is and how the Enhanced Service Discovery works

Reviewer: ```suggestion
Learn what is and how the Kubernetes Service Discovery for Workflow Application Configuration works
```

[.card]
--
[.card-title]

> xref:cloud/deploying-on-minikube.adoc[Enhanced Service Discovery]
> [.card-description]
> Learn what is and how the Enhanced Service Discovery works
> --
> 
> [.card]
> --
> [.card-title]
> xref:cloud/enhanced-service-discovery.adoc[Deploying your {context} application on Minikube]

Reviewer 2: ```suggestion
xref:cloud/enhanced-service-discovery.adoc[Enhanced Service Discovery]
[.card-description]
Learn what is and how the Enhanced Service Discovery works
--

[.card]
--
[.card-title]
xref:cloud/deploying-on-minikube.adoc[Deploying your {context} application on Minikube]
```
